File: tests/rspec/pages/design/template_group_forms.rb
class TemplateGroupCreate < ControlPanelPage
  set_url_matcher /design\/group\/create/

  element :save_button, '.form-btns-top input.btn'

  element :name, 'input[name=group_name]'
  elements :duplicate_existing_group, 'div[data-input-value="duplicate_group"] input[type="radio"]'
  element :is_site_default, 'a[data-toggle-for="is_site_default"]'

  def load
    self.open_dev_menu
    click_link 'Templates'
    find('a[href*="cp/design/group/create"]').click
  end
end

class TemplateGroupEdit < ControlPanelPage
  set_url_matcher /design\/group\/edit/

  element :save_button, '.form-btns-top input.btn'

  element :name, 'input[name=group_name]'
  element :is_site_default, 'a[data-toggle-for="is_site_default"]'

  def load_edit_for_group(name)
    self.open_dev_menu
    click_link 'Templates'
    find('a[href*="cp/design/group/edit/' + name + '"]').click
  end
end

File: tests/cypress/cypress/elements/pages/design/TemplateGroupForms.js
import ControlPanel from '../ControlPanel'

class TemplateGroupCreate extends ControlPanel {
    constructor() {
        super()
        this.url = 'admin.php?/cp/design/group/create';

        this.elements({
            "save_button": '.form-btns-top input.btn',
            "name": 'input[type=text][name=group_name]',
            "duplicate_existing_group": 'div[data-input-value="duplicate_group"] input[type="radio"]',
            "is_site_default": 'a[data-toggle-for="is_site_default"]',
        })
    }
}

class TemplateGroupEdit extends ControlPanel {
    constructor() {
        super()
        this.url = 'admin.php?/cp/design/group/edit';

        this.elements({
            "save_button": '.form-btns-top input.btn',
            "name": 'input[type=text][name=group_name]',
            "is_site_default": 'a[data-toggle-for="is_site_default"]',
        })
    }
    load_edit_for_group(name) {
        cy.visit(`${this.url}/${name}`)
    }
}
export default { TemplateGroupCreate, TemplateGroupEdit }
// export default TemplateGroupCreate;